packages/styled-system/src/utils/useInjectGlobalStyles.tsx

```diff
@@ -0,0 +1,20 @@
+import { isNil } from "../utils/assertion.ts";
+import { useInsertionEffectWithFallback } from "./useInsertionEffectWithFallback.ts";
+
+export function useInjectGlobalStyles(elementId: string, cssContent: string) {
```

Comment:
I'm not a fan of this function because i don't like the usage. But it prevent some code duplication. Open to suggestion